codegen/transforms/svc_interface.sysl

```diff
@@ -93,7 +93,10 @@ CodeGenTransform:
         let tn = ep.value.ret where (.key != "error" && .value != "ok" && .value != "error") -> <sequence of string>(t:
           out = makeType(t.value, package).out
         )
-        TypeName = Join(tn flatten(.out) | ["error"], ", ")
+        let ok = ep.value.ret where (.value == "ok") -> <sequence of string>(t:
```

Comment:
I have a feeling that headers are returned in another situation. Is this correct?

Returning headers for an unknown type seems a little arbitrary. I would have assumed that the successful result type would be either `*http.Response` (so the caller is free to inspect the entire response) or `interface{}` (to simply indicate a successful response).

Reply:
This just fixes a part that should have been taken care when we started accommodating for return <: ok. The intent is to surely reengineer return <: ok support but not necessarily as part of this bug fix.

Reply:
I see. So the previous intention (right or wrong) was that an untyped return would result in an `*http.Headers` result, and this pull request just fixes a related issue. This is fine.

Reply:
@andrewemeryanz yes, returning the full restlib.HTTPResponse would be much better